### PR TITLE
Subscribe block

### PR DESCRIPTION
**Pull request by sophieH29**
**# Desktop**

subscribe_desktop1

subscribe_desktop2

subscribe_desktop3

subscribe_desktop4

**# Mobile**

subscribe_mobile1

subscribe_mobile2

subscribe_mobile5

### CODE REVIEW

File: scripts/redux/actions.js [hunk 1]
```diff
@@ -137,3 +137,110 @@ const teamActions = {
...
+            iconURL: signInObject.user.photoURL,
+            provider: 'https://accounts.google.com'
+          });
+
+          navigator.credentials.store(cred);
+        }
+
+        store.dispatch({
+          type: SIGN_IN,
+          user: Object.assign(signInObject.user, { signedIn: true })
```
Comment (ozasadnyy) [inline scripts/redux/actions.js]: in this case you modify `signInObject.user`. To create a new one do:
```
Object.assign({}, signInObject.user, { signedIn: true})
```
or spread operator
```
{
 ...signInObject.user,
 signedIn: true
}
```

File: scripts/redux/actions.js [hunk 2]
```diff
@@ -137,3 +137,110 @@ const teamActions = {
       });
   }
 };
+
+const userActions = {
+  signIn: () => {
+    let provider = new firebase.auth.GoogleAuthProvider();
```
Comment (ozasadnyy) [inline scripts/redux/actions.js]: Customers asked to add other sign in option. Could we do that?

File: scripts/redux/actions.js [hunk 3]
```diff
@@ -137,3 +137,110 @@ const teamActions = {
...
+        });
+      });
+  },
+
+  autoSignIn: () => {
+    let currentUser = firebase.auth().currentUser;
+    if (currentUser) {
+      store.dispatch({
+        type: AUTO_SIGN_IN,
+        user: Object.assign(currentUser, { signedIn: true })
```
Comment (ozasadnyy) [inline scripts/redux/actions.js]: same

File: scripts/redux/constants.js [hunk 4]
```diff
@@ -27,3 +28,11 @@ const FETCH_SPEAKERS_LIST = 'FETCH_SPEAKERS_LIST';
 const FETCH_GALLERY = 'FETCH_GALLERY';
 // team
 const FETCH_TEAM = 'FETCH_TEAM';
+// sign in
+const SIGN_IN = 'SIGN_IN';
+// sign out
```
Comment (ozasadnyy) [inline scripts/redux/constants.js]: the idea of this comments is to divide by re reduces, so you can remove this and next comment

File: src/elements/subscribe-dialog.html [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+             }
+        }
+
+        _closeDialog() {
+            dialogsActions.closeDialog(DIALOGS.SUBSCRIBE);
+        }
+
+        _resetFields() {
+             this.validate = false;
+             this.shadowRoot.querySelector('#firstNameInput').value = "";
```
Comment (ozasadnyy) [inline src/elements/subscribe-dialog.html]: use single quotes

File: scripts/redux/actions.js (hunk 2 above)
Comment (sophieH29) [inline scripts/redux/actions.js]: ok, I will add

File: scripts/redux/actions.js (hunk 1 above)
Comment (sophieH29) [inline scripts/redux/actions.js]: spread operator is good one :)

Comment (ozasadnyy) [inline scripts/redux/actions.js]: Yeah, but currently is not supported in most browser, so maybe just before the release we will have to change it back to `Object.assign` :/

File: database.rules.json [hunk 6]
```diff
@@ -21,6 +21,10 @@
     "speakers": {
       ".read": true,
       ".write": "root.child('users').child(auth.uid).child('rules').child('speakers').val() === true"
+    },
+    "subscribers": {
+       ".read": false,
+       ".write": true
```
Comment (ozasadnyy) [inline database.rules.json]: In this case, everyone will be able to write and edit other users data

File: src/elements/signin-dialog.html [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+        .icon-twitter {
+            color: var(--twitter-color);
+        }
+
+        .icon-facebook {
+            color: var(--facebook-color);
+        }
+      
+
+        @media (min-width: 640px) {
```
Comment (ozasadnyy) [inline src/elements/signin-dialog.html]: remove empty block

File: src/elements/signin-dialog.html [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+        static get properties() {
+            return {                
+                subscribed: {
+                    type: Boolean,
+                    statePath: 'subscribed'
+                }
+            }
+        }  
+
+        googleSignIn() {
```
Comment (ozasadnyy) [inline src/elements/signin-dialog.html]: it's would be better create one generic function and get provider value from attributes

File: database.rules.json (hunk 6 above)
Comment (sophieH29) [inline database.rules.json]: how that would be possible? We shoul allow to write here, even if user isn't login

File: src/elements/signin-dialog.html (hunk 8 above)
Comment (sophieH29) [inline src/elements/signin-dialog.html]: I wantet, bu event on-tap does not return target element which was clicked :(

File: src/elements/signin-dialog.html (hunk 7 above)
Comment (sophieH29) [inline src/elements/signin-dialog.html]: thanks :) Acually, I left it just to see if you would notice :dancing_women:

File: database.rules.json (hunk 6 above)
Comment (ozasadnyy) [inline database.rules.json]: hmm, good point. In this case, leave as it is

File: src/elements/signin-dialog.html (hunk 8 above)
Comment (ozasadnyy) [inline src/elements/signin-dialog.html]: Add Polymer.GestureEventListeners
https://www.polymer-project.org/2.0/docs/devguide/gesture-events

Comment (sophieH29) [inline src/elements/signin-dialog.html]: will do, thanks

**Pull request merged**